File: packages/native-bridge/plugins/share-in/ios/Sources/ShareInPlugin.swift
import Foundation
import Capacitor

/**
 * A minimal stub implementation of the ShareIn plugin for iOS.  This
 * class exposes a single method, `getSharedFiles`, which currently
 * returns an empty array.  To implement share-sheet integration you
 * would create a Share Extension target in Xcode and save incoming
 * files to a shared App Group.  The main app can then read the
 * shared files and return them via this plugin.
 */
@objc(ShareInPlugin)
public class ShareInPlugin: CAPPlugin {
    @objc public func getSharedFiles(_ call: CAPPluginCall) {
        call.resolve(["files": []])
    }
}

File: packages-disabled/native-bridge/plugins/share-in/android/src/main/java/com/cleanshare/sharein/ShareInPlugin.java
package com.cleanshare.sharein;

import com.getcapacitor.JSArray;
import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.annotation.CapacitorPlugin;
import com.getcapacitor.PluginMethod;

/**
 * A minimal stub implementation of the ShareIn plugin for Android.  The
 * purpose of this class is to expose a JavaScript API while
 * development of the native share-sheet integration is underway.  The
 * getSharedFiles() method currently resolves with an empty array but
 * demonstrates how to construct return objects.  You can extend
 * this class to implement receiving ACTION_SEND intents and
 * forwarding URIs into the app’s sandbox.
 */
@CapacitorPlugin(name = "ShareIn")
public class ShareInPlugin extends Plugin {
    @PluginMethod
    public void getSharedFiles(PluginCall call) {
        JSArray files = new JSArray();
        JSObject ret = new JSObject();
        ret.put("files", files);
        call.resolve(ret);
    }
}